### PR TITLE
[BackEnd] Add Books pagination/filtering Feature

### LINKED ISSUE #25: Either Add/Change API call for Books to Include Sorting and Filtering
The front end will send a request for some number of books with some type of filter/sorting. The back end will return this number of books for the query.

### LINKED ISSUE #27: Change Book List API to Support Multiple Pages
Rather than just returning the top books, we will need to be able to return a range of books for a given filter/sort query

### LINKED ISSUE #45: Create API Endpoint for Obtaining Number of Books in a Given Genre
The front end will need this for the genre list page

### PR DESCRIPTION
**Pull request by hkder**
## Feature Description (what did you do?)
- Book List Sortable by Any of the Displayed Fields (GET) - /api/v1/books/?ordering=field 
(Closes #25)
- Book List Filtered by Selectable Genre (GET) - /api/v1/books/?genre=genre (Closes #45)
- Book List Filtered by Keyword Search (GET) - /api/v1/books/?search=keyword
- Book Pagination can be set (GET) - /api/v1/books/?page=n&page_size=m (Closes #27)

## Design/Implementation Details (how did you do it?)
- Created Pagination Rule - paginations.py
- Added pagination, filtering, ordering - views.py
- Transforms query set based on filter and ordering - views.py (ListCreateBookAPIView)

## Areas Affected (what parts of the code does this affect?)
- Frontend API Endpoints for Filtering/Sorting Books
- Frontend API Endpoints for Pagination

## Testing (how did you ensure this works correctly?)
- Postman Book-API section on [imaginary-books.postman.co](https://imaginary-books.postman.co/workspace/Team-Workspace~aee197b8-0b74-4c72-8e7e-8964df269e2c/overview)

## Future Considerations (is there anything we should know about this going forward?)
- Add Genre, Retail Price Suggestions
- Currently supports multi-genre for a book but might not need the feature

### CODE REVIEW

File: backend/books/views.py [hunk 1]
```diff
@@ -88,6 +95,16 @@ def getOrCreateModel(self, item_list, model):
...
                 name=item.strip(),
             )
+    
+    def get_queryset(self):
+        default_query_set = Book.objects.all()
+        default_query_set = default_query_set.annotate(author=Max('authors__name'))
+        default_query_set = default_query_set.annotate(genre=Max('genres__name'))
+
+        if genre := self.request.query_params.get('genre'):
+            return default_query_set.filter(genres__name=genre)
```
Comment (danielfeinblatt) [inline backend/books/views.py:105]: Would you mind just explaining what these line are doing? Did some research, but a bit confused, especially about the `Max` function being used. And what is the if statement for/why does it only apply to genres?

**Review by danielfeinblatt — COMMENTED** [review]
Just left one clarifying question. Once answered, I'll give it a final approval

File: backend/books/views.py (hunk 1 above)
Comment (hkder) [reply backend/books/views.py]: just added a comment to annotations and genre filtering

**Review by danielfeinblatt — APPROVED** [review]
LGTM

**Pull request merged**